### PR TITLE
'3rSolution'

### PR DESCRIPTION
**Pull request by ketstap162**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,67 @@
...
+        self.km = km
+
+    def __str__(self):
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self):
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other):
+        if type(other) == type(self):
```
Comment (AriohBelskij) [inline app/main.py:12]: I think you should use isinstance() here, as in every method where you compare types. In any case, it will look more beautiful

**Review by AriohBelskij — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 2]
```diff
@@ -15,7 +15,7 @@ def __add__(self, other):
             return Distance(self.km + other)
 
     def __iadd__(self, other):
-        if type(other) == type(self):
+        if isinstance(other, type(self)):
```
Comment (AriohBelskij) [inline app/main.py:18]: u need use isinstance in all ur methods! and u dont need else.

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,67 @@
...
+
+    def __iadd__(self, other):
+        if isinstance(other, type(self)):
+            self.km += other.km
+        else:
+            self.km += other
+        return self
+
+    def __mul__(self, other):
+        if type(other) == type(self):
```
Comment (AriohBelskij) [inline app/main.py:25]: use isinstance() and remove else:

File: app/main.py [hunk 4]
```diff
@@ -1,3 +1,67 @@
...
+            return Distance(self.km * other.km)
+        else:
+            return Distance(self.km * other)
+
+    def __truediv__(self, other):
+        if type(other) == type(self):
```
Comment (AriohBelskij) [inline app/main.py:31]: use isinstance() and remove else:

File: app/main.py [hunk 5]
```diff
@@ -1,3 +1,67 @@
...
+            return Distance(round(self.km / other.km, 2))
+        else:
+            return Distance(round(self.km / other, 2))
+
+    def __lt__(self, other):
+        if type(other) == type(self):
```
Comment (AriohBelskij) [inline app/main.py:37]: use isinstance() and remove else:

File: app/main.py [hunk 6]
```diff
@@ -1,3 +1,67 @@
...
+            return self.km < other.km
+        else:
+            return self.km < other
+
+    def __gt__(self, other):
+        if type(other) == type(self):
```
Comment (AriohBelskij) [inline app/main.py:43]: use isinstance() and remove else:

File: app/main.py [hunk 7]
```diff
@@ -1,3 +1,67 @@
...
+            return self.km > other.km
+        else:
+            return self.km > other
+
+    def __eq__(self, other):
+        if type(other) == type(self):
```
Comment (AriohBelskij) [inline app/main.py:49]: use isinstance() and remove else:

File: app/main.py [hunk 8]
```diff
@@ -1,3 +1,67 @@
...
+            return self.km == other.km
+        else:
+            return self.km == other
+
+    def __le__(self, other):
+        if type(other) == type(self):
```
Comment (AriohBelskij) [inline app/main.py:55]: use isinstance() and remove else:

File: app/main.py [hunk 9]
```diff
@@ -1,3 +1,67 @@
...
+            return self.km <= other.km
+        else:
+            return self.km <= other
+
+    def __ge__(self, other):
+        if type(other) == type(self):
```
Comment (AriohBelskij) [inline app/main.py:61]: use isinstance() and remove else:

**Review by AriohBelskij — CHANGES_REQUESTED** [review]
fix it

**Comment by ketstap162** [comment]
> fix it

I fixed it.
What about else... I think it should be.

**Comment by AriohBelskij** [comment]
> > fix it
> 
> I fixed it. What about else... I think it should be.

return ends the function. u can write like a - 
if isinstance(666, int) :
       return some what u want if isinstance()
return some what u want if not isinstance()

U dont need else. Fix it

File: app/main.py [hunk 10]
```diff
@@ -9,7 +9,7 @@ def __repr__(self):
         return f"Distance(km={self.km})"
 
     def __add__(self, other):
-        if type(other) == type(self):
+        if isinstance(other, type(self)):
             return Distance(self.km + other.km)
         else:
```
Comment (AriohBelskij) [inline app/main.py:14]: remove else)

**Review by AriohBelskij — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 11]
```diff
@@ -11,8 +11,7 @@ def __repr__(self):
     def __add__(self, other):
         if isinstance(other, type(self)):
```
Comment (AriohBelskij) [inline app/main.py:12]: explicitly show the type, now type(self) looks very confusing, specify the class like - isinstance(other, Distance)
change it in all ur methods ;D

**Review by AriohBelskij — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 11 above)
Comment (AriohBelskij) [reply app/main.py:12]: and fix flake8

**Pull request closed without merging**